### PR TITLE
Migrate to netcoreapp2.0

### PR DESCRIPTION
**Pull request by Tratcher**
Some NTLM tests are disabled due to a client issue: https://github.com/aspnet/ServerTests/issues/82

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj [hunk 1]
```diff
@@ -13,14 +13,11 @@
...
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Hosting" Version="$(AspNetCoreVersion)" />
-    <PackageReference Include="Microsoft.Extensions.RuntimeEnvironment.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
     <PackageReference Include="Microsoft.Extensions.TaskCache.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
     <PackageReference Include="Microsoft.Net.Http.Headers" Version="$(AspNetCoreVersion)" />
-    <PackageReference Include="System.Threading.Overlapped" Version="$(CoreFxVersion)" />
-  </ItemGroup>
-
-  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
+    <PackageReference Include="Microsoft.Win32.Registry" Version="$(CoreFxVersion)" />
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj]: Hmm?

File: src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj [hunk 2]
```diff
@@ -13,14 +13,11 @@
...
     <PackageReference Include="System.Security.Principal.Windows" Version="$(CoreFxVersion)" />
+    <PackageReference Include="System.Threading.Overlapped" Version="$(CoreFxVersion)" />
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj]: Are these already in M.NETCore.App?

File: src/Microsoft.AspNetCore.Server.HttpSys/fx/Microsoft/Win32/SafeHandles/CriticalHandleZeroOrMinusOneIsInvalid.cs [hunk 3]
```diff
@@ -1,29 +0,0 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
-
-#if NETSTANDARD1_3
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Server.HttpSys/fx/Microsoft/Win32/SafeHandles/CriticalHandleZeroOrMinusOneIsInvalid.cs]: 👏

File: test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs [hunk 4]
```diff
@@ -169,8 +166,7 @@ public class AuthenticationTests
             }
         }
 
-        [ConditionalFact]
-        [SkipOffDomain]
+        [ConditionalFact(Skip = "Requires a domain joined machine")]
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Server.HttpSys.FunctionalTests/Listener/AuthenticationTests.cs]: Tracking item?

File: src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj (hunk 1 above)
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj]: Yup, Http.Sys reads registry settings. :-)
Previously it was doing this only on net46.

File: src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj (hunk 2 above)
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/Microsoft.AspNetCore.Server.HttpSys.csproj]: I was able to remove System.Threading.Overlapped, but Windows and Registry are still required.

File: src/Microsoft.AspNetCore.Server.HttpSys/NativeInterop/ComNetOS.cs [hunk 5]
```diff
@@ -15,11 +15,7 @@ static ComNetOS()
         {
             var win8Version = new Version(6, 2);
 
-#if NETSTANDARD1_3
-            IsWin8orLater = (new Version(RuntimeEnvironment.OperatingSystemVersion) >= win8Version);
-#else
             IsWin8orLater = (Environment.OSVersion.Version >= win8Version);
```
Comment (Eilon) [inline src/Microsoft.AspNetCore.Server.HttpSys/NativeInterop/ComNetOS.cs]: How do you know you're running on Windows, though? What does this do in Linux 8.3?

Comment (Eilon) [inline src/Microsoft.AspNetCore.Server.HttpSys/NativeInterop/ComNetOS.cs]: Or if this is somehow not relevant here, maybe add a comment that says we know we're on Windows if we get here??

Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.HttpSys/NativeInterop/ComNetOS.cs]: The Windows check happens much earlier:
https://github.com/aspnet/HttpSysServer/blob/50ed4564f1c2fe936a43358d268b89fe7d32f018/src/Microsoft.AspNetCore.Server.HttpSys/HttpSysListener.cs#L56

Haven't actually tried it on mac/linux, I expect it to be pretty explosive.

Comment (Eilon) [inline src/Microsoft.AspNetCore.Server.HttpSys/NativeInterop/ComNetOS.cs]: Ok.

**Pull request merged**